packages/jaeger-ui/src/utils/DeepDependencyGraph/parse-payload.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+type TPayload = {
+  operation: string;
+  service: string;
+}[][];
+
+type TService = {
+  name: string;
+  operations: Record<string, TOperation>;
+};
+
+type TOperation = {
+  name: string;
+  pathElems: TPathElem[];
+  service: TService;
+};
+
+type TPathElem = {
+  memberOf: TPath;
+  operation: TOperation;
+  pathIdx: number;
+  visibilityIdx: number;
+};
+
+type TPath = {
+  focalIdx: number;
+  members: TPathElem[];
+};
+
+type TServiceMap = Record<string, TService>;
```

Comment:
Good call, updated. Just a minor issue with typescript.
```
if (!serviceMap.has(serviceName)) {
  serviceMap.set(serviceName, { /* ... */ });
}
if (!serviceMap.get(serviceName).operations.has(operationName)) {
  /* ... */
}
```
TypeScript complains that `serviceMap.get(serviceName)` could be undefined, but of course it can't be.